lib/ensureNano.js

```diff
@@ -0,0 +1,52 @@
+function ensureNano(nano, db) {
+    var nanoDb = nano.use(db);
+    return wrapNano(nano, nanoDb, db);
+}
+
+function wrapNano(nano, nanoDb, db) {
+    var nanoObject = {};
+    Object.keys(nanoDb).forEach(function(method) {
+        if (typeof nanoDb[method] == "function") {
+            return nanoObject[method] = function() {
+                ensureDatabase.call(this, nano, nanoDb, db, method, arguments)
+            };
+        }
+
+        if (typeof nanoDb[method] == "object") {
+            return nanoObject[method] = wrapNano(nano, nanoDb[method], db);
+        }
+
+        return nanoObject[method] = nanoDb[method];
+    });
+
+    return nanoObject;
+}
+
+function ensureDatabase(nano, nanoDb, db, method, args) {
```

Comment:
This function is tricky to follow.

I would expect the calls to the original nanoDb to be: `nanoDb[method].apply(nanoDb, args);` since you want them executed with the original context.